### PR TITLE
Use correct abbreviations for data quantities. Fixes #13054

### LINKED ISSUE #13054: Trino CLI reports data with wrong units
Using the Trino CLI to query CSV data via the Hive connector, I noticed that the CLI is reporting the amount of data transferred incorrectly. Querying a 72.8 MB CSV file, the CLI reported that it had transferred only 69.4 MB of data.

It seems that the discrepancy arises from the fact that macOS, correctly, reports file sizes in kilobytes, megabytes, gigabytes, etc, based on powers of 1,000, with the abbreviations KB, MB, GB, while the Trino CLI is reporting the number of kibibytes, mebibytes, gibibytes etc, based on powers of 1,024. but using the KB/MB/GB abbreviations rather than the correct KiB/MiB/GiB. The math confirms this - (69.4 * 1,024 * 1,024) / (1,000 * 1,000) = 72.8.

See [this wikipedia page](https://en.wikipedia.org/wiki/Megabyte) for an explanation of the correct terms to use.

The Trino CLI should either use the KiB/MiB/GiB abbreviations or, preferably, report the number of kilobytes/megabytes/gigabytes correctly.

### PR DESCRIPTION
**Pull request by metadaddy**
## Description

The Trino CLI currently reports data size incorrectly, using the legacy, 1,024-based, values for K, M, G etc rather than the modern 1,000-based values.

> Is this change a fix, improvement, new feature, refactoring, or other?

Fix

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

CLI

## Non-technical explanation

This change corrects the display of data sizes in the CLI to use KiB, MiB for 1,024-based values and adds a command option, `--decimal-data-size` to display data sizes in 1,000-based units.

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

The CLI now shows data sizes and rates with binary (1024-based) abbreviations: KiB, MiB, etc. A new command option, `--decimal-data-size`, shows decimal (1000-based) values and abbreviations: KB, MB, etc. (#13054)

## Related issues, pull requests, and links

* Fixes #13054
* Replaces #13095

### CODE REVIEW

**Comment by electrum** [comment]
Since Trino uses base-2 units everywhere, I think it would be better to keep the base-2 units and but add the `i` to the unit designation. This preserves the existing behavior and compatibility with base-2 units used everywhere else. If we make the CLI use base-10, then it will show different values than other places in Trino while using the same suffix.

The downside of adding the suffix is that it will break all of the carefully formatted output, since the values will now be wider. But as long as we still fit in 80 characters, it should be ok.

We could also add a non-default command line option to use base-10 units, so as not to change the existing behavior. You can use a [configuration file](https://trino.io/docs/current/client/cli.html#configuration-file) to effectively make this a default for your CLI installation.

**Comment by metadaddy** [comment]
The impact on the formatting of the CLI is minimal:

```
trino:ds> select count(*) from drivestats;
   _col0   
-----------
 346006813 
(1 row)

Query 20221129_225226_00008_9q4rt, FINISHED, 1 node
Splits: 420 total, 420 done (100.00%)
12.27 [346M rows, 4.66MiB] [28.2M rows/s, 389KiB/s]
```

Same query with `--decimal-data-size `:

```
trino:ds> select count(*) from drivestats;
   _col0   
-----------
 346006813 
(1 row)

Query 20221129_225715_00012_9q4rt, FINISHED, 1 node
Splits: 421 total, 421 done (100.00%)
11.02 [346M rows, 4.89MB] [31.4M rows/s, 443KB/s]
```

**Comment by hashhar** [comment]
@electrum Your comments seem to be addressed. Can you please take a look?

File: client/trino-cli/src/test/java/io/trino/cli/TestFormatUtils.java [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+ */
+package io.trino.cli;
+
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestFormatUtils
```
Comment (hashhar) [inline client/trino-cli/src/test/java/io/trino/cli/TestFormatUtils.java:22]: Thanks for adding this.

**Review by hashhar — APPROVED** [review]
LGTM % @electrum's review.

**Comment by mosabua** [comment]
:wave: @metadaddy  - this PR has become inactive. We hope you are still interested in working on it. Please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

Also cc @electrum and @hashhar

**Comment by hashhar** [comment]
Thanks for reminder @mosabua. Sorry @metadaddy that this fell through the cracks.

I think we can just rebase this and merge. I'll wait a few days for @metadaddy if he does the rebase otherwise I'll do a rebase and merge.

**Comment by metadaddy** [comment]
@hashhar Hey there - apologies for the latency in replying! I can rebase if you like - it should be relatively straightforward.

**Comment by metadaddy** [comment]
Something messed up here. I'll figure it out and re-push!

**Comment by metadaddy** [comment]
@hashhar I'm not sure why it's checking commit 0d2d2e92998ee2c8bce483cbbc6f2898c25e9309. It should be checking c944f126a4f788e845381b8384cd70ddbea9fc1a. Is there something I can do on my side to fix this?

**Comment by metadaddy** [comment]
@hashhar Looks like everything is good to go.

**Comment by mosabua** [comment]
Could you add the new flag to the documentation @metadaddy  .. if timing for merge doesnt work out I can review a separate follow up PR as well

**Comment by metadaddy** [comment]
Hi @mosabua - I just added the flag to the doc as an amendment to the "Add option for decimal data size/rates, change KB etc to KiB etc for binary" commit: [18b9780](https://github.com/trinodb/trino/pull/14859/commits/18b978085a0de6ab726993fced0c60fd52eef415) 👍🏻

**Comment by metadaddy** [comment]
Hi @mosabua - I just added the flag to the doc as an amendment to the "Add option for decimal data size/rates, change KB etc to KiB etc for binary" commit: [18b9780](https://github.com/trinodb/trino/pull/14859/commits/18b978085a0de6ab726993fced0c60fd52eef415) 👍🏻

**Comment by mosabua** [comment]
Great .. I pinged @hashhar and @electrum for review

**Review by hashhar — APPROVED** [review]
LGTM. Default is unchanged and is opt-in.

Only affects CLI. `format` related functions in Trino still behave same as before.

**Review by mosabua — APPROVED** [review]

**Comment by mosabua** [comment]
I agree with the assessment from @hashhar 

All concerns from @electrum  are addressed. 

I will merge. Worst case we can do a follow up.

**Pull request merged**